funding/manager.go

```diff
@@ -2849,17 +3280,97 @@ func (f *Manager) handleFundingLocked(peer lnpeer.Peer,
 
 	// First, we'll attempt to locate the channel whose funding workflow is
 	// being finalized by this message. We go to the database rather than
-	// our reservation map as we may have restarted, mid funding flow.
+	// our reservation map as we may have restarted, mid funding flow. Also
+	// provide the node's public key to make the search faster.
 	chanID := msg.ChanID
-	channel, err := f.cfg.FindChannel(chanID)
+	channel, err := f.cfg.FindChannel(peer.IdentityKey(), chanID)
 	if err != nil {
 		log.Errorf("Unable to locate ChannelID(%v), cannot complete "+
 			"funding", chanID)
 		return
 	}
 
+	// We'll need to store the received TLV alias if the option_scid_alias
+	// feature was negotiated. This will be used to provide route hints
+	// during invoice creation. In the zero-conf case, it is also used to
+	// provide a ChannelUpdate to the remote peer. This is done before the
+	// call to InsertNextRevocation in case the call to PutPeerAlias fails.
+	// If it were to fail on the first call to handleFundingLocked, we
+	// wouldn't want the channel to be usable yet.
+	if channel.NegotiatedAliasFeature() {
+		// If the AliasScid field is nil, we must fail out. We will
+		// most likely not be able to route through the peer.
+		if msg.AliasScid == nil {
+			log.Debugf("Consider closing ChannelID(%v), peer "+
+				"does not implement the option-scid-alias "+
+				"feature properly", chanID)
+			return
+		}
+
+		// We'll store the AliasScid so that invoice creation can use
+		// it.
+		err = f.cfg.AliasManager.PutPeerAlias(chanID, *msg.AliasScid)
+		if err != nil {
+			log.Errorf("unable to store peer's alias: %v", err)
+			return
+		}
+
+		// If we do not have an alias stored, we'll create one now.
+		// This is only used in the upgrade case where a user toggles
+		// the option-scid-alias feature-bit to on. We'll also send the
+		// funding_locked message here in case the link is created
+		// before sendFundingLocked is called.
+		aliases := f.cfg.AliasManager.GetAliases(
+			channel.ShortChannelID,
+		)
+		if len(aliases) == 0 {
+			// No aliases were found so we'll request and store an
+			// alias and use it in the funding_locked message.
+			alias, err := f.cfg.AliasManager.RequestAlias()
+			if err != nil {
+				log.Errorf("unable to request alias: %v", err)
+				return
+			}
+
+			err = f.cfg.AliasManager.AddLocalAlias(
+				alias, channel.ShortChannelID, false,
+			)
+			if err != nil {
+				log.Errorf("unable to add local alias: %v",
+					err)
+				return
+			}
+
+			secondPoint, err := channel.SecondCommitmentPoint()
```

Comment:
Why can't we just use `NextRevocationKey` here?

Reply:
We can here since the channel can't have advanced state, but I just made everything use the same function

Reply:
Oh my bad. We may have actually advanced state at this point, so this is necessary.